### DISCUSSION TITLE
RaspberryMatic unter Apple M1+

### DISCUSSION THREAD

**Discussion by krobipd** [OpenCCU/OpenCCU#2723] [🇩🇪 Fragen&Antworten] [answered]
Hallo

ich versuche seit tagen mit allen möglichen Mitteln irgendeine version der ccu auf meinem m1 zum laufen zu bekommen.
hab utm, parallels, vm ware alles probiert.

das ovm image geht gar nicht, nur das aarch64 lies sich einpaar mal mit utm im abgesicherten modus starten - mehr aber auch nicht. selbst wenn ich eine vm aufmache mit Debian und dann debmatic, pivccu oder RaspberryMatic versuche zu installieren startet das ganze nicht.

der maximalste erfolg bis jetzt war der splash-screen von debmatic bzw pivccu. aber die reghass starte nie - soweit bin ich schon mal. hab aber schon gelesen das wohl die aarch64 bin wohl lauffähig wäre auf einem m1. ich bitte um info wie wo was 

bitte danke

**Comment by jens-maus** [comment, marked as answer]
Hierzu (RaspberryMatic unter Apple M1, etc. Prozessoren) gab es in verschiedensten Foren bereits reichlich Diskussionen. Siehe dazu hier:

https://homematic-forum.de/forum/viewtopic.php?f=65&t=82312#p802188
https://homematic-forum.de/forum/viewtopic.php?f=65&t=74823&p=725240#p725133

Kurze Zusammenfassung: Den Apple ARM Chips (Apple Silicon, Apple M1, etc.) fehlen die notwendigen 32bit Extensions (wurde von Apple eingespart) um gewisse 32bit Software noch ausführen zu können. Und da RaspberryMatic auch noch 32bit binaries mit sich bringt wird das leider bis auf weiteres nichts werden mit RaspberryMatic unter Apple Silicon. Abgesehen davon braucht es das aber IMHO eh nicht wirklich, da es eigentlich auch überhaupt keinen Sinn ergibt eine Hausautomation die 24/7 laufen muss unter einem solchen System laufen zu lassen.

**Reply by krobipd** [reply]
vielen lieben dank für die info, ich werde auf alle fälle mal weiterschauen wie ich es hinbekomme. nativ wohl nicht. ich weiß über den sinn und zwecks lässt sich herrlich diskutieren ;-)

**Comment by krobipd** [comment]
hallo,

wie darf man die aktuellen Entwicklungen den deuten? neue arm64 binaries usw heißt dass das es doch Hoffnung gibt auf eine native arm64 raspberrymatic?

**Reply by jens-maus** [reply]
Die Hoffnung stirbt bekanntlich immer zuletzt. Es ist aber in der Tat zumindest ein kleiner weiterer Schritt in Richtung einer kompletten 64bit Version von RaspberryMatic. Noch ist es aber noch nicht soweit und somit wird das auch weiterhin nichts mit Apple Silicon und virtuelles RaspberryMatic. Da gibt es immer noch eine handvoll 32bit binaries (die leider auch noch closed source sind) die weiterhin das verhindern. Es bleibt also dabei: Geht nicht. Brauch man aber auch nicht, denn es ist ohnehin etwas over-the-top RaspberryMatic auf einem teuren Apple Silicon laufen lassen zu wollen. Einfach einen Pi nehmen oder nen günstigen x86 mit proxmox nutzen um darunter RaspberryMatic als x86_64 vm zu betreiben und schon geht alles damit...

**Reply by krobipd** [reply]
wie ist den der status mit der aktuellen version? da hat sich ja einiges getan bei den binaries.

**Reply by jens-maus** [reply]
Kannst du gerne nochmal mit der 3.83.6.20250705 probieren ob die generic-aarch64 version von RaspberryMatic nun auf einem Apple Silicon basierten System/Umgebung nun geht. Kann sein das die Basisfunktionalität nun geht. Jedoch wirst du dann mitunter vielleicht wieder auf eine Limitation stoßen sobald du z.B. Addons wie CUxD verwenden willst die es aktuell immer noch nur als 32bit Versionen gibt und daher auch diese weiterhin nicht auf einem Apple Silicon basiertem System lauffähig sein werden.

**Reply by krobipd** [reply]
thx, ja das sind ja grundsätzlich gute Neuigkeiten :-) nach deinem tollen release wird es nur eine Frage der Zeit sein muss auch cuxd 64bit arm Binaries haben wird :-) (hoff ich mal) danke sehr

**Comment by jpkraemer** [comment]
Läuft bei mir schon ewig problemlos in UTM auf einem M1 Mac. Dass Apple Silicon für 24/7 nicht geeignet wäre würde ich so nicht unterschreiben, vorher liefen die Services alle auf einem NAS, das hat für deutlich weniger Performance und deutlich weniger Workloads fast 10 mal so viel Strom verbraucht...

**Reply by meistermolli** [reply]
hi @jpkraemer, wie hast du das dann ans laufen bekommen? Kannst du die VM näher beschreiben was hast du für ein System eingerichtet? Ist ein HmIP-RFUSB angesteckt den du dann durchreichst? Gibt es eine Möglichkeit das automatisch zu machen oder muss man das nach jedem Mac Neustart wieder neu durchreichen?



@jens-maus deine Arbeit und Lösungen sind für jeden Anwender gold wert einfach Danke dafür!!! Ohne raspberrymatic würde ich garkein HmIP mehr nutzen. Natürlich verliert sie ein stück weit ja ihre Berechtigung und den Namen mit einer Umstellung auf die vm's. Aber die Welt ist mittlerweile eine andere.

Auf lange Sicht wäre natürlich ein HomeAssistant Addon super was eine selber angelegte VM oder selbst laufenden raspi ablöst. Natürlich ist ein M1 oder alles was folgt oversized. Aber ich bin absolut zufrieden so. Auf einem Mini M4 habe ich eine VM mit homeassistant, alles weitere als addon darin installiert. Das Ansprechverhalten ist so schnell das selbst die Unifi Kameras die mit Scrypted an Homeassistant gehen und dann weiter in HomeKit chneller (mehrere ms) eine Pushbenachrichtigung schicken wer da vor der Haustür steht (inkl Namen und Gesichtserkennung) wie die Unifi Protect app selbst. Das einzige gerät was ich aktuell "extra" benötige ist der alte Rpi 3b+ der schon jahre im Rack vor sich hin läuft. Wenn der irgendwann mal den geist auf gibt wäre es cool auf eine Digitale Lösung umzusteigen.

Performanter, Backup Sicherer und Redundanter könnte das ganze System (was sich heutzutage dann nicht mehr allein um hmip dreht) dann nicht mehr sein.
